internal/pkg/skuba/kubernetes/node_version_test.go

```diff
@@ -18,394 +18,1027 @@
+			fakeClientset: fake.NewSimpleClientset(
+				&corev1.NodeList{
+					Items: []corev1.Node{
+						createFakeMasterNode(master0, clusterVersion, clusterVersion, false),
+						createFakeMasterNode(master1, wrongVersion, clusterVersion, false),
+					},
 				},
-				"my-worker-1": {
-					Nodename:                 "my-worker-1",
-					ContainerRuntimeVersion:  version.MustParseSemantic("1.13.1"),
-					KubeletVersion:           version.MustParseSemantic("1.13.1"),
-					APIServerVersion:         nil,
-					ControllerManagerVersion: nil,
-					SchedulerVersion:         nil,
-					EtcdVersion:              nil,
-					Unschedulable:            true,
+				&corev1.PodList{
+					Items: []corev1.Pod{
+						// master0
+						createFakePod(apiserver, master0, clusterVersion),
+						createFakePod(controllerManager, master0, clusterVersion),
+						createFakePod(scheduler, master0, clusterVersion),
+						createFakePod(etcd, master0, clusterVersion),
+						// master1
+						createFakePod(apiserver, master1, wrongVersion),
+						createFakePod(controllerManager, master1, clusterVersion),
+						createFakePod(scheduler, master1, clusterVersion),
+						createFakePod(etcd, master1, clusterVersion),
+					},
 				},
-			},
+			),
+			expectReturn: false,
 		},
 	}
 	for _, tt := range nodes {
 		tt := tt // Parallel testing
 		t.Run(tt.name, func(t *testing.T) {
-			allWorkerNodesTolerateVersionWithVersioningInfo := allWorkerNodesTolerateVersionWithVersioningInfo(tt.nodeVersionInfoMap, tt.currentClusterVersion)
-			if !reflect.DeepEqual(allWorkerNodesTolerateVersionWithVersioningInfo, tt.expectedResult) {
-				t.Errorf("got %v, want %v", allWorkerNodesTolerateVersionWithVersioningInfo, tt.expectedResult)
+			actualReturn, err := AllControlPlanesMatchVersion(tt.fakeClientset, cluster)
+			if err != nil {
+				t.Errorf("error not expected: (%v)", err)
+				return
+			}
+			if actualReturn != tt.expectReturn {
+				t.Errorf("got: (%v), want: (%v)", actualReturn, tt.expectReturn)
 			}
 		})
 	}
 }
 
-func TestAllControlPlanesMatchVersionWithVersioningInfo(t *testing.T) {
+func TestAllNodesMatchClusterVersionWithVersioningInfo(t *testing.T) {
+	clusterVersion := version.MustParseSemantic("1.14.1")
+	wrongVersion := version.MustParseSemantic("1.13.1")
 	var nodes = []struct {
-		name                    string
-		currentClusterVersion   *version.Version
-		expectedResult          bool
-		containerRuntimeVersion string
-		nodeVersionInfoMap      NodeVersionInfoMap
+		name               string
+		nodeVersionInfoMap NodeVersionInfoMap
+		expectReturn       bool
 	}{
 		{
-			name:                  "all control planes match",
-			currentClusterVersion: version.MustParseSemantic("1.14.1"),
-			expectedResult:        true,
+			name: "all nodes match cluster version",
+			nodeVersionInfoMap: NodeVersionInfoMap{
+				master0: {
+					Nodename:         master0,
+					KubeletVersion:   clusterVersion,
+					APIServerVersion: clusterVersion,
+					Unschedulable:    false,
```

Comment:
Guess you want to test `Unschedulable` behavior ;)

Reply:
Redundant, removed.